conn/bind_windows.go

```diff
@@ -473,32 +484,38 @@ func (bind *afWinRingBind) Send(buf []byte, nend *WinRingEndpoint, isOpen *atomi
 	return winrio.SendEx(bind.rq, dataBuffer, 1, nil, addressBuffer, nil, nil, 0, 0)
 }
 
-func (bind *WinRingBind) Send(buf []byte, endpoint Endpoint) error {
+func (bind *WinRingBind) Send(buffs [][]byte, endpoint Endpoint) error {
 	nend, ok := endpoint.(*WinRingEndpoint)
 	if !ok {
 		return ErrWrongEndpointType
 	}
 	bind.mu.RLock()
 	defer bind.mu.RUnlock()
-	switch nend.family {
-	case windows.AF_INET:
-		if bind.v4.blackhole {
-			return nil
-		}
-		return bind.v4.Send(buf, nend, &bind.isOpen)
-	case windows.AF_INET6:
-		if bind.v6.blackhole {
-			return nil
+	for _, buf := range buffs {
```

Comment:
SendEx takes a data buffer count parameter. Rather than calling SendEx(1) in a loop, shouldn't all of these be batched at once, so that SendEx(n) is called instead?

Reply:
Unfortunately that API only accepts a count of 0, or a count of 1, despite seemingly implying otherwise. Furthermore if it did accept more buffers, the behavior would likely be more similar to scatter/gather (producing one large packet) rather than transmitting multiple packets. Here the buffers being sent are packets, and so need to be sent independently.

The API we would use for sending multiple packets at once is TransmitPackets (https://learn.microsoft.com/en-us/windows/win32/api/mswsock/nc-mswsock-lpfn_transmitpackets?redirectedfrom=MSDN). This API comes with new ABI as well https://learn.microsoft.com/en-us/windows/win32/api/mswsock/ns-mswsock-transmit_packets_element - as mentioned above, we want to get to this but would prefer to do it in a separate change.

Reply:
Ah bingo. Okay. Do you guys have plans to actually do that ever?

Reply:
Yes, soon. Windows is important to us too.